workers/task-driver/src/tasks/node_startup.rs

```diff
@@ -333,12 +340,27 @@ impl NodeStartupTask {
             info!("elected leader is a cluster peer");
             self.task_state = NodeStartupTaskState::JoinRaft;
         } else {
-            self.task_state = NodeStartupTaskState::SetupRelayerWallet;
+            self.task_state = NodeStartupTaskState::CreateGlobalMatchingPool;
         }
 
         Ok(())
     }
 
+    /// Create the global matching pool
+    async fn create_global_matching_pool(&self) -> Result<(), NodeStartupTaskError> {
+        // If the node recovered from a snapshot, we do not need to re-create the global
+        // matching pool
+        if self.state.was_recovered_from_snapshot() {
+            return Ok(());
+        }
+
+        // Otherwise, create the global matching pool
+        let waiter = self.state.create_global_matching_pool().await?;
```

Comment:
This should probably just be done in [setup_node_metadata](https://github.com/renegade-fi/renegade/blob/caa46562c7ae1e636d033feb6a4ffb524ca03b73/state/src/interface/node_metadata.rs#L104) (maybe we can rename it to `setup_node_state` or something. It doesn't really make sense to send this through consensus in my opinion. Given that it's static information, the best way to achieve consensus is to put it in code!

Reply:
hmm, fair point, yeah. will make this change.